econml/_ortho_learner.py

```diff
@@ -739,15 +713,17 @@ def fit(self, Y, T, *, X=None, W=None, Z=None, sample_weight=None, freq_weight=N
 
         if not only_final:
 
-            if self.binary_outcome:
+            if self.discrete_outcome:
                 self.outcome_transformer = LabelEncoder()
                 self.outcome_transformer.fit(Y)
                 if Y.shape[1:] and Y.shape[1] > 1:
                     raise ValueError(
-                        f"Only one outcome variable is supported when binary_outcome=True. Got Y of shape {Y.shape}")
+                        f"Only one outcome variable is supported when discrete_outcome=True. Got Y of shape {Y.shape}")
                 if len(self.outcome_transformer.classes_) > 2:
                     raise AttributeError(
-                        "More than 2 outcome classes detected. This method currently only supports binary outcomes")
+                        f"({self.outcome_transformer.classes_} outcome classes detected. \
```

Comment:
I think you're including the classes themselves rather than their count here.

Reply:
Good catch